code/test/sixsq/nuvla/server/resources/nuvlabox_2_lifecycle_test.clj

```diff
@@ -465,7 +465,7 @@
               (-> session
                   (request cluster-nuvlabox
                            :request-method :post
-                           :body (json/write-str {:cluster-action "join-worker" :nuvlabox-manager-status {}}))
+                           :body (json/write-str {:cluster-action "join-worker" :nuvlabox-manager-status "{}"}))
```

Comment:
This line is strange. The body is a json string with cluster-action and nuvlabox-manager-status keys. But nuvlabox-manager-status value is a string with with following content "{}"

Reply:
that is to be consistent with the limitation at https://github.com/nuvla/ui/blob/master/code/src/cljs/sixsq/nuvla/ui/edge_detail/views.cljs#L337
, since it always sends nuvlabox-manager-status as a string

Reply:
Okey. There is an issue on the UI then.
```
(when-not (empty? (:nuvlabox-manager-status @form-data))
  (swap! form-data assoc :nuvlabox-manager-status (general-utils/edn->json :nuvlabox-manager-status @form-data))))
```

I don't really get the meaning of these lines.
when not empty :nuvlabox-manager-status in form-data atom, transform nuvlabox-manager-status to json string.

Nowing that UI use clojure nuvla client library. form-data will be passed will be transformed from edn to json before sending the request. So operation body will be something like: {"cluster-action": "join-worker", "nuvlabox-manager-status": "{...}"}

Why do you need nuvlabox-manager-status in json string format nested inside another json string body request?

Reply:
I don't, it should be an object, but the UI is sending a string instead

Reply:
I think on the UI just send the form-data and all his fields in edn format. It will be transformed to json string by clojure-library. 
On api-server, body is transformed back to edn by the middleware.